api/routes/subcriterias/get_sub_criteria.py

```diff
@@ -43,3 +49,160 @@ def return_subcriteria_from_config(sub_criteria_id):
+        ]
+
+    @classmethod
+    def convert_boolean_values(cls, themes_answers: list[dict]) -> list[dict]:
+        """function checks boolean values in themes_answers & replace
+        boolean values to string. (False -> "No", True -> "Yes")
+        Args:
+            themes_answers (_type_): array of dict
+        """
+        current_app.logger.info("Converting boolean values to strings")
+        for answers in themes_answers:
+            if "answer" in answers.keys():
+                if answers["answer"] == False:
+                    answers.update(answer="No")
+                if answers["answer"] == True:
+                    answers.update(answer="Yes")
+
+    @classmethod
+    def sort_add_another_component_contents(
+        cls, themes_answers: list[dict]
+    ) -> list[dict]:
+        """function checks for special presentation_type "heading"
+        in array of themes answers, if exists, adds question-value
+        to an answer key for presentation_type "heading" theme and
+        looks for presentation_type of "description" and "amount" with
+        same presentation_type "heading"s field_id, retrieve words only
+        from strings of answers for presentation_type "description" theme
+        & numbers only for presentation_type "amount" theme.
+
+        Args:
+            themes_answers (_type_): array of dict
+        """
+        for heading in themes_answers:
+            try:
+                if heading["presentation_type"] == "heading":
+                    current_app.logger.info(
+                        "mapping add-another component contents"
+                    )
+                    heading["answer"] = heading["question"]
+                    for theme in themes_answers:
+                        if (
+                            heading["field_id"] in theme.values()
+                            and theme["presentation_type"] == "description"
+                        ):
+                            description_answer = [
+                                re.sub("[^a-zA-Z]+", " ", description)
+                                for description in theme["answer"]
+                            ]
+                            theme["answer"] = description_answer
+
+                        if (
+                            heading["field_id"] in theme.values()
+                            and theme["presentation_type"] == "amount"
+                        ):
+                            amount_answer = [
+                                re.sub("[^\d+.]", "", amount)
+                                for amount in theme["answer"]
+                            ]
+
+                            theme["answer"] = amount_answer
+
+                        else:
+                            continue
+
+            except (KeyError, IndexError):
+                current_app.logger.error("Incorrect field key")
+
+    @classmethod
+    def map_grouped_fields_answers(cls, theme: dict, questions: dict) -> list:
```

Comment:
I'm a bit confused about the return type here, isn't this returning a `tuple` rather than a `list`? Maybe I'm misreading the code?

Reply:
Yes, you are right. It should be a tuple. Forgot to update. Fixed!

Reply:
Got rid of regex & now using split method to get the amount